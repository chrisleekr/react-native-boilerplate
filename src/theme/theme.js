import {Dimensions} from 'react-native';
import {colors} from './colors';
const {width, height} = Dimensions.get('window');

// Themes:- Can alter values here. Can only be consumed through Context (see useTheme.js file)
const themes = {
  light: {
    name: 'light',
    barStyle: 'light-content',
    color: '#000000',
    labelColor: '#525E75',
    invertColor: '#ffffff',
    layoutBg: '#f0f0f0',
    headerBg: '#f6f7f8',
    cardBg: '#e4dfa7',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#B00020',
    buttonActiveColor: '#6200EE',
    buttonDisabledColor: 'rgba(0, 0, 0, 0.26)',
    card: {
      shadowColor: '#000000',
      borderColor: '#B00020'
    },
    ...colors
  },
  dark: {
    name: 'dark',
    barStyle: 'dark-content',
    color: '#ffffff',
    labelColor: '#F1DDBF',
    invertColor: '#000000',
    layoutBg: '#121212',
    headerBg: '#1e1e1e',
    cardBg: '#3f4045',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#ffffff',
    buttonActiveColor: '#BB86FC',
    buttonDisabledColor: 'rgba(255,255,255, 0.26)',
    card: {
      shadowColor: '#121212',
      borderColor: '#ffffff'
    },
    ...colors
  }
};

const window = {
  // app dimensions
  width,
  height
};

export {themes, window};
