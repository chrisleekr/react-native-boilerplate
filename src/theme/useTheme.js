/**
 * For Dark/Light Mode
 * @exports ThemeContext, ThemeProvider, useTheme
 */

import React, {useState, createContext, useContext} from 'react';
import {Appearance} from 'react-native';

// Import preconfigured themes from theme file
import {themes} from './theme';

// Context
export const ThemeContext = createContext(null);

// Provider to be used in index/App/or top of any parent
export const ThemeProvider = ({children}) => {
  const colorScheme = Appearance.getColorScheme();

  const [theme, setTheme] = useState(themes[colorScheme]);
  return (
    <ThemeContext.Provider value={{theme, setTheme}}>
      {children}
    </ThemeContext.Provider>
  );
};

// useTheme hook for easy access to theme and setTheme
export const useTheme = () => {
  const {theme, setTheme} = useContext(ThemeContext);

  const toggleTheme = v => {
    setTheme(v ? themes.dark : themes.light);
  };

  return {theme, toggleTheme};
};
