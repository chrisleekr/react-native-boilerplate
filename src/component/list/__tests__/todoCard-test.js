import 'react-native';
import React from 'react';

import {cleanup, fireEvent, render} from '@testing-library/react-native';

import {ThemeProvider} from '../../../theme/useTheme';

import TodoCard from '../todoCard';

let rendered;
let element;

let todo;

const leftIconOnPress = jest.fn();
const rightIconOnPress = jest.fn();

describe('TodoCard', () => {
  afterEach(cleanup);

  beforeEach(() => {
    todo = {
      state: 'ongoing',
      name: 'My todo'
    };
    rendered = render(
      <ThemeProvider>
        <TodoCard
          loading={false}
          leftIconOnPress={leftIconOnPress}
          leftIconName={'checkbox-blank-circle-outline'}
          leftIconSize={20}
          leftIconColor="#000000"
          todo={todo}
          rightIconOnPress={rightIconOnPress}
          rightIconName="trash-can-outline"
          rightIconSize={20}
          rightIconColor="#000000"
        />
      </ThemeProvider>
    );
  });

  it('matches snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  describe('left-icon', () => {
    beforeEach(() => {
      element = rendered.getByTestId('left-icon');
    });

    it('rendered left-icon', () => {
      expect(element).toBeTruthy();
    });

    it('sets left-icon as enabled', () => {
      expect(element.props.accessibilityState).toStrictEqual({disabled: false});
    });

    describe('leftIconOnPress', () => {
      beforeEach(() => {
        fireEvent(element, 'onPress');
      });

      it('triggers onPress', () => {
        expect(leftIconOnPress).toHaveBeenCalled();
      });
    });
  });

  describe('text', () => {
    beforeEach(() => {
      element = rendered.getByText('My todo');
    });

    it('rendered text', () => {
      expect(element).toBeTruthy();
    });
  });

  describe('right-icon', () => {
    beforeEach(() => {
      element = rendered.getByTestId('right-icon');
    });

    it('rendered right-icon', () => {
      expect(element).toBeTruthy();
    });

    it('sets right-icon as enabled', () => {
      expect(element.props.accessibilityState).toStrictEqual({disabled: false});
    });

    describe('rightIconOnPress', () => {
      beforeEach(() => {
        fireEvent(element, 'onPress');
      });

      it('triggers onPress', () => {
        expect(rightIconOnPress).toHaveBeenCalled();
      });
    });
  });

  describe('todo with completed', () => {
    beforeEach(() => {
      todo = {
        state: 'completed',
        name: 'My completed todo'
      };
      rendered = render(
        <ThemeProvider>
          <TodoCard
            loading={false}
            leftIconOnPress={leftIconOnPress}
            leftIconName={'checkbox-marked-circle'}
            leftIconSize={20}
            leftIconColor="#000000"
            todo={todo}
            rightIconOnPress={rightIconOnPress}
            rightIconName="trash-can-outline"
            rightIconSize={20}
            rightIconColor="#000000"
          />
        </ThemeProvider>
      );
    });

    it('matches snapshot', () => {
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  });

  describe('loading is true', () => {
    beforeEach(() => {
      todo = {
        state: 'ongoing',
        name: 'My todo'
      };
      rendered = render(
        <ThemeProvider>
          <TodoCard
            loading={true}
            leftIconOnPress={leftIconOnPress}
            leftIconName={'checkbox-blank-circle-outline'}
            leftIconSize={20}
            leftIconColor="#000000"
            todo={todo}
            rightIconOnPress={rightIconOnPress}
            rightIconName="trash-can-outline"
            rightIconSize={20}
            rightIconColor="#000000"
          />
        </ThemeProvider>
      );
    });

    it('matches snapshot', () => {
      expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('sets left-icon as disabled', () => {
      expect(
        rendered.getByTestId('left-icon').props.accessibilityState
      ).toStrictEqual({disabled: true});
    });

    it('sets right-icon as disabled', () => {
      expect(
        rendered.getByTestId('right-icon').props.accessibilityState
      ).toStrictEqual({disabled: true});
    });
  });
});
