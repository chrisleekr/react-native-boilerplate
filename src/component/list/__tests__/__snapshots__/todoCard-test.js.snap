// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard loading is true matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "space-between",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 10,
      },
      Object {
        "backgroundColor": "#e4dfa7",
        "shadowColor": "#000000",
      },
    ]
  }
  testID="todo-card"
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexBasis": 25,
        "flexGrow": 0,
        "flexShrink": 0,
        "opacity": 1,
      }
    }
    testID="left-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰄰
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "flexGrow": 1,
          "flexShrink": 0,
          "fontSize": 15,
        },
        Object {
          "color": "#000000",
        },
        Object {},
      ]
    }
    testID="text"
  >
    My todo
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="right-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰩺
    </Text>
  </View>
</View>
`;

exports[`TodoCard matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "space-between",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 10,
      },
      Object {
        "backgroundColor": "#e4dfa7",
        "shadowColor": "#000000",
      },
    ]
  }
  testID="todo-card"
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexBasis": 25,
        "flexGrow": 0,
        "flexShrink": 0,
        "opacity": 1,
      }
    }
    testID="left-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰄰
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "flexGrow": 1,
          "flexShrink": 0,
          "fontSize": 15,
        },
        Object {
          "color": "#000000",
        },
        Object {},
      ]
    }
    testID="text"
  >
    My todo
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="right-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰩺
    </Text>
  </View>
</View>
`;

exports[`TodoCard todo with completed matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "space-between",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 10,
      },
      Object {
        "backgroundColor": "#e4dfa7",
        "shadowColor": "#000000",
      },
    ]
  }
  testID="todo-card"
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexBasis": 25,
        "flexGrow": 0,
        "flexShrink": 0,
        "opacity": 1,
      }
    }
    testID="left-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰄳
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "flexGrow": 1,
          "flexShrink": 0,
          "fontSize": 15,
        },
        Object {
          "color": "#525E75",
        },
        Object {
          "textDecorationLine": "line-through",
        },
      ]
    }
    testID="text"
  >
    My completed todo
  </Text>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="right-icon"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰩺
    </Text>
  </View>
</View>
`;
